docs/expressions/parsing.md

```diff
@@ -22,9 +22,17 @@ math.evaluate([expr1, expr2, expr3, ...], scope)
 
 Function `evaluate` accepts a single expression or an array with
 expressions as the first argument and has an optional second argument
-containing a scope with variables and functions. The scope can be a regular
-JavaScript Object, or Map. The scope will be used to resolve symbols, and to write
-assigned variables or function.
+containing a `scope` with variables and functions. The scope can be a regular
+JavaScript `Map` (recommended) or `Object`. The scope will be used to resolve 
```

Comment:
Should it be mentioned that it can also be a custom class implementing a Map-like interface (and define this interface)? These docs seem to mention that only a native Map or Object ({}) is supported, while the example code actually shows `CustomMap` and `AdvancedCustomMap` (that's exactly what I'm talking about). However, it obviously depends on whether you want to support and encourage such usage.

Reply:
Thanks, this is useful feedback. I've refined the explanation about scopes here to clearly mention JavaScript objects and custom classes implementing a `Map` interface.